src/js/components/FileInput/FileInput.js

```diff
@@ -124,46 +121,109 @@ const FileInput = forwardRef(
     // rightPad needs to be included in the rightOffset
     // otherwise input may cover the RemoveButton, making it
     // unreachable by mouse click.
+    // If browse anchor or button is greater then remove button then
+    // rightoffset will take the larger width
     let rightOffset;
-    if (removeRef.current) {
+    if (removeRef.current && controlRef.current) {
+      const rightOffsetBrowse = controlRef.current.getBoundingClientRect()
+        .width;
+      const rightOffsetRemove = removeRef.current.getBoundingClientRect().width;
       if (rightPad && typeof rightPad === 'string')
         rightOffset =
           removeRef.current.getBoundingClientRect().width +
           rightPad.replace('px', '');
-      else rightOffset = removeRef.current.getBoundingClientRect().width;
+      if (files.length === 1 || files.length > aggregateThreshold) {
+        rightOffset =
+          controlRef.current.getBoundingClientRect().width +
```

Comment:
since we created variables above, should we just use rightOffsetBrowse and rightOffsetRemove in all of these blocks instead of calculating them again